### PR TITLE
feat(router): [OpenNode] response reference id

### LINKED ISSUE #2342: [FEATURE]: [OpenNode] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by eigengravy**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Closes #2342.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by eigengravy** [comment]
Closes #2342

File: crates/router/src/connector/opennode/transformers.rs [hunk 1]
```diff
@@ -114,7 +114,7 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(item.response.data.id),
```
Comment (srujanchikke) [inline crates/router/src/connector/opennode/transformers.rs:117]: Hello @eigengravy,

The `connector_response_reference_id` should be mapped to the merchant reference ID of the connector response, which is `order_id` according to the API documentation of OpenNode. Please add an optional field called `order_id` in the payment response and use it here.

Thank you.

Comment (srujanchikke) [reply crates/router/src/connector/opennode/transformers.rs:117]: Please refer api documentation of OpenNode [here](https://developers.opennode.com/reference/create-charge)

Comment (eigengravy) [reply crates/router/src/connector/opennode/transformers.rs]: Made the required changes.

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

Please update the branch to ensure the spell check in the CI workflow functions properly.

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @eigengravy!

**Comment by eigengravy** [comment]
Have pulled the latest commits. Thank you.

**Comment by VedantKhairnar** [comment]
Hey @eigengravy  :wave:,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**